### ISSUE TITLE
Using custom braces in message template does not work

### ISSUE THREAD

**Issue by xNinjaKittyx** [PyCQA/pylint#5636]
### Bug description

Have any list of errors:

On pylint 1.7 w/ python3.6 - I am able to use this as my message template
```
$ pylint test.py --msg-template='{{ "Category": "{category}" }}'
No config file found, using default configuration
************* Module [redacted].test
{ "Category": "convention" }
{ "Category": "error" }
{ "Category": "error" }
{ "Category": "convention" }
{ "Category": "convention" }
{ "Category": "convention" }
{ "Category": "error" }
```

However, on Python3.9 with Pylint 2.12.2, I get the following:
```
$ pylint test.py --msg-template='{{ "Category": "{category}" }}'
[redacted]/site-packages/pylint/reporters/text.py:206: UserWarning: Don't recognize the argument '{ "Category"' in the --msg-template. Are you sure it is supported on the current version of pylint?
  warnings.warn(
************* Module [redacted].test
" }
" }
" }
" }
" }
" }
```

Is this intentional or a bug?

### Command used

```shell
pylint test.py --msg-template='{{ "Category": "{category}" }}'
```

### Pylint output

```shell
[redacted]/site-packages/pylint/reporters/text.py:206: UserWarning: Don't recognize the argument '{ "Category"' in the --msg-template. Are you sure it is supported on the current version of pylint?
  warnings.warn(
************* Module [redacted].test
" }
" }
" }
" }
" }
" }
```

### Expected behavior

Expect the dictionary to print out with `"Category"` as the key.

### Pylint version

```shell
Affected Version:
pylint 2.12.2
astroid 2.9.2
Python 3.9.9+ (heads/3.9-dirty:a2295a4, Dec 21 2021, 22:32:52) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]


Previously working version:
No config file found, using default configuration
pylint 1.7.4, 
astroid 1.6.6
Python 3.6.8 (default, Nov 16 2020, 16:55:22) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

**Comment by xNinjaKittyx** [comment]
Subsequently, there is also this behavior with the quotes
```
$ pylint test.py --msg-template='"Category": "{category}"'
************* Module test
Category": "convention
Category": "error
Category": "error
Category": "convention
Category": "convention
Category": "error

$ pylint test.py --msg-template='""Category": "{category}""'
************* Module test
"Category": "convention"
"Category": "error"
"Category": "error"
"Category": "convention"
"Category": "convention"
"Category": "error"
```

**Comment by xNinjaKittyx** [comment]
Commit that changed the behavior was probably this one: https://github.com/PyCQA/pylint/commit/7c3533ca48e69394391945de1563ef7f639cd27d#diff-76025f0bc82e83cb406321006fbca12c61a10821834a3164620fc17c978f9b7e

And I tested on 2.11.1 that it is working as intended on that version.

**Comment by Pierre-Sassoulas** [comment]
Thanks for digging into this !
